### PR TITLE
Add match_class! macro

### PR DESCRIPTION
**Pull request by OddCoincidence**
See #771 for more details and discussion.

### CODE REVIEW

File: vm/src/macros.rs [hunk 1]
```diff
@@ -147,3 +147,90 @@ macro_rules! extend_class {
...
+///     f @ PyFloat => f.to_f64().to_bigint().unwrap(),
+///     obj => panic!("non-numeric object {}", obj)
+/// );
+///
+/// assert!(int_value.is_zero());
+/// ```
+#[macro_export]
+macro_rules! match_class {
+    // The default arm.
+    ($obj:expr, _ => $default:expr) => {
```
Comment (coolreader18) [inline vm/src/macros.rs]: Could probably add a `$(,)?` here to allow trailing commas.

File: vm/src/macros.rs [hunk 2]
```diff
@@ -147,3 +147,90 @@ macro_rules! extend_class {
         )*
     }
 }
+
+/// Macro to match on the built-in class of a Python object.
+///
+/// Like `match`, `match_object!` is an expression, and so must have a default
+/// arm that will be evaluated when the object was not an instance of any of
+/// the classes specified in other arms.
```
Comment (coolreader18) [inline vm/src/macros.rs]: Should be `match_class!`, right?

Comment (coolreader18) [inline vm/src/macros.rs]: Also, is there any way to enable a missing default path? `match` is allowed as just a statement, where everything just leads to a `()` type. Unless this is to ~sorta emulate algebraic types and requiring that all cases are covered? In which case I'd recommend making the documentation clearer here.

Comment (OddCoincidence) [reply vm/src/macros.rs]: Yeah it needs to be exhaustive like `match`, I'll improve the docs

**Pull request merged**